### PR TITLE
Fix nested jest test and add coverage for method

### PR DESCRIPTION
**Pull request by edgibbs**
Found an issue with a broken build caused by a nested test.  Fixes that and cleaned up test coverage for one method.

### CODE REVIEW

File: app/javascript/Clients/clientIdPage/ClientIdPage.test.js [hunk 1]
```diff
@@ -10,6 +10,16 @@ describe('Client ID Page', () => {
     clientPage = shallow(<ClientIdPage />);
   });
 
+  describe('#handleSelect', () => {
+    it('stops propogation', () => {
+      const event = { stopPropagation: () => {} };
+      const stopPropagation = jest.spyOn(event, 'stopPropagation');
+      const wrapper = mount(<ClientIdPage />).instance();
+      wrapper.handleSelect('_href', event);
+      expect(event.stopPropagation).toHaveBeenCalled();
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/ClientIdPage.test.js]: Just wondering if there's a cleaner way to do this, my mocking experience from Jasmine doesn't quite map over to Jest.

**Comment by eckdanny-osi** [comment]
Can confirm this resolves the *failing* tests introduced in #137

```
 FAIL  app/javascript/Clients/clientIdPage/ClientInformation.test.js
  Client Information
    ✓ renders Card Component (4ms)
    ✓ renders InputComponent components (3ms)
    ✓ renders CheckboxRadioGroup components (2ms)
    #handleChange()
      ✓ should manage the selection (5ms)
    #handleDropDownChange()  function
      ✓ should an event handler that sets state (2ms)
      ✓ toggles true when checked CSEC data (3ms)
      ✕ toggles the display of the csec block  (6ms)

  ● Client Information › #handleDropDownChange()  function › toggles the display of the csec block

    Tests cannot be nested. Test `dropdown is available on click` cannot run because it is nested within `toggles the display of the csec block `.

      at Env.it (../../node_modules/jest-jasmine2/build/jasmine/Env.js:428:15)
      at Object.<anonymous> (Clients/clientIdPage/ClientInformation.test.js:59:7)
          at new Promise (<anonymous>)
          at <anonymous>
```

**Comment by eckdanny-osi** [comment]
The Warnings (introduced in #121) are still present, but not required to resolve in this PR. 

```
 PASS  app/javascript/Clients/clientIdPage/ClientInformation.test.js
  Client Information
    ✓ renders Card Component (10ms)
    ✓ renders InputComponent components (3ms)
    ✓ renders CheckboxRadioGroup components (8ms)
    #handleChange()
      ✓ should manage the selection (5ms)
    #handleDropDownChange()  function
      ✓ should an event handler that sets state (4ms)
      ✓ toggles true when checked CSEC data (2ms)
      ✓ toggles the display of the csec block  (7ms)
      ✓ dropdown is available on click (7ms)

  console.error node_modules/fbjs/lib/warning.js:33
    Warning: validateDOMNesting(...): Text nodes cannot appear as a child of <tbody>.
        in tbody (created by Table)
        in table (created by Table)
        in div (created by Table)
        in Table (created by Ethnicity)
        in div (created by Ethnicity)
        in div (created by Ethnicity)
        in div (created by Cards)
        in div (created by Cards)
        in div (created by Cards)
        in Cards (created by Ethnicity)
        in Ethnicity (created by ClientIdPage)
        in div (created by ClientIdPage)
        in div (created by ClientIdPage)
        in div (created by ClientIdPage)
        in div (created by ClientIdPage)
        in ClientIdPage (created by WrapperComponent)
        in WrapperComponent
```

**Comment by edgibbs** [comment]
Yeah the other warning bugs me, but Melbourne and I spent a few minutes looking into it yesterday, and we didn't figure out the exact cause.

**Comment by eckdanny-osi** [comment]
Not a matter for this PR, but the naming conventions in the eventHandler chain are not particularly sensical... here's what i mean:

I thought:

1. _Why is the method being tested named `handleSelect`?_, and
1. _Why is a positional argument `href` present that doesn't appear to do anything?_

Some digging through wood-duck raised my following conclusion:

The eventHandling chain is: 

1. `ClientIdPage#handleSelect` -->
1. `ClientIdSidebar#handleSelect` -->
1. `NavLink#clickHandler` (wood duck) -->
1. `Link#clickHandler` (wood duck) -->
1. `a#onClick`

The [`a#onClick` callback](https://github.com/ca-cwds/react-wood-duck/blob/master/src/Link.js#L12)  (for some reason) is applied to args`(href: String, event: Event)`. 

It is unfortunate that consistency with DOM apis has been broken:

 > When the event handler is specified as an HTML attribute, the specified code is wrapped into a function with the following parameters: `(event: Event)` for all event handlers, except `onerror`. 

If one needs the `href` property for anchors: `event.target.href` (its already on the event)

**Comment by eckdanny-osi** [comment]
I've had success in the past sticking to a simple and repeatable convention for naming event handlers:

 - The bound component method is named `handle<Event>`
 - The prop (html attribute) binding is named `on<Event>`

**Comment by eckdanny-osi** [comment]
I'll merge. Will open a sep issue for warnings mentioned in https://github.com/ca-cwds/case-management/pull/142#issuecomment-357008527 and assign to author of #121 for resolution.

**Pull request merged**